src/averageAges.js

```diff
@@ -55,7 +71,13 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => onlyWithSon
+    ? people.find(woman => woman.name === person.mother) && person.sex === 'm'
+    : people.find(woman => woman.name === person.mother));
```

Comment:
DRY
make a separate function for it
```suggestion
    ? hasMom(...) && person.sex === 'm'
    : hasMom(...);
```